From 8c0efc280fbfb4b38cabe56060e265ab19f0d890 Mon Sep 17 00:00:00 2001
Message-Id: <8c0efc280fbfb4b38cabe56060e265ab19f0d890.1427001998.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Wed, 18 Mar 2015 02:50:01 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: check for chv and pipe B enabled
 planes

This patch adds a check for chv and updates watermarks only
if pm5 and DDR DVFS feature is supported. This also corrects
the logic for checking enabled planes on PIPE B.

This patch needs to be squashed with
Author: vandita kulkarni <vandita.kulkarni@intel.com>
Date:   Fri Feb 27 17:33:52 2015 +0530

    FOR_UPSTREAM [VPG]: drm/i915: Support Pm5 and DDR DVFS on CHV
	Change-Id: I766ef0edff29843951e5312f6fd0b9a08963162b

Issue: GMINL-4468
Change-Id: Ie4b7f1a7bf1718ad70d746aa78c556f1fd4a36a4
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++--
 drivers/gpu/drm/i915/intel_pm.c      |    2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 52155f4..9ce08e0 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11152,8 +11152,9 @@ static int intel_crtc_set_display(struct drm_crtc *crtc,
 	/* Commit to registers */
 	ret = intel_set_disp_commit_regs(disp, dev, intel_crtc);
 
-	/* updat the watermarks*/
-	valleyview_update_wm_pm5(crtc);
+	/* PM5 requires that we update the watermarks straight way */
+	if (IS_CHERRYVIEW(dev))
+		valleyview_update_wm_pm5(crtc);
 
 	/* Enable maxfifo if needed */
 	intel_update_maxfifo(dev_priv, intel_crtc->pipe, plane_cnt);
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 0e9a25c..ffb6b77 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1559,7 +1559,7 @@ void intel_update_maxfifo(struct drm_i915_private *dev_priv, enum pipe pipe,
 	 * enabled.
 	 */
 	if (single_plane_enabled(dev_priv->plane_stat
-			& (PIPE_A_PLANES_MASK || PIPE_B_PLANES_MASK))
+			& (PIPE_A_PLANES_MASK | PIPE_B_PLANES_MASK))
 		&& !(dev_priv->plane_stat & PIPE_C_MASK)
 		&& (single_plane_enabled(dev_priv->plane_stat
 			& PIPE_ENABLE_MASK))
-- 
1.7.9.5

